core/objects.go

```diff
@@ -361,6 +397,57 @@ type Certificate struct {
 	Expires time.Time  `db:"expires"`
 }
 
+func (cert Certificate) MatchesCSR(csr *x509.CertificateRequest, earliestExpiry time.Time) (err error) {
```

Comment:
Please add a comment above this that explains what this method does, and what checks are being done.